### ISSUE TITLE
Bong smartwatch

### ISSUE THREAD

**Issue by diimdeep** [micooke/arduino-nRF5-smartwatches#6]
I have 'Bong smartwatch' with nRF52832

![2017-02-20 23-30-34](https://user-images.githubusercontent.com/485713/41065439-551123f4-69e7-11e8-863c-ea8e2ba0fa02.JPG)
![2017-02-20 23-31-50](https://user-images.githubusercontent.com/485713/41065440-554545c6-69e7-11e8-8c18-aa00fa567648.jpg)
![2017-02-20 23-48-53](https://user-images.githubusercontent.com/485713/41065441-5568d572-69e7-11e8-940f-f6ad3fb838e1.JPG)

Doable?

**Comment by micooke** [comment]
Yeah very doable. The DIO and CLK pins are labelled, there are pads for UART TX RX for debugging, and VCC and GND for power.
Having said all this, you now need to reverse engineer the pinout. I would start by using the NRF52 generic from sandeeps repo, and then take my SoftwareI2C library and use the scanner example to find the pins (obs you need to cycle through pin options) and take the SoftwareSerial and cycle through pins to find the TX and RX pad to pin mapping. The software serial lib requires you to patch sandeeps repo with a PR I have with Sandeep. That's a start anyway

**Comment by micooke** [comment]
So assuming you want to go ahead with this, the first thing I will say is that the process is not reverseable - you can't get back to the original firmware.
You will likely need to unlock the nrf first, see  https://github.com/micooke/arduino-nRF5-smartwatches/blob/master/nrf52_disable_read_protection.txt

Once you have done that you need to flash the firmware (using any nrf52 variant from sandeeps repo). Then you should be able to program it normally. The first thing you will note is that the vibration motor is going crazy. You need to drive the vibration motor pin high (just throw everything high to begin with). The right pin needs to be OUTPUT_PULLUP but that is not an option for sandeeps repo.

**Comment by micooke** [comment]
Oh, and it looks like the same screen as the T28 variant I have. I doubt it has the same pinout, but you never know. The oled pinout will be the trickiest, you will need to buzz that out with a very small probe. I hacked a cheap volt meter to use a header pin tip, then made a couple attachments that end in a single braided wire. I found this cheaper and a lot less rough to the pcb than using a proper precision probe tip
